src/test/java/com/marek/onlinebookstore/config/CustomMySqlContainer.java

```diff
@@ -0,0 +1,32 @@
+package com.marek.onlinebookstore.config;
+
+import org.testcontainers.containers.MySQLContainer;
+
+public class CustomMySqlContainer extends MySQLContainer<CustomMySqlContainer> {
+    public static final String DB_IMAGE = "mysql:8.4.0";
+
+    private static CustomMySqlContainer mySqlContainer;
+
+    public CustomMySqlContainer() {
+        super(DB_IMAGE);
+    }
+
+    public static synchronized CustomMySqlContainer getInstance() {
+        if (mySqlContainer == null) {
+            mySqlContainer = new CustomMySqlContainer();
+        }
+        return mySqlContainer;
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        System.setProperty("TEST_DB_URL", mySqlContainer.getJdbcUrl());
+        System.setProperty("TEST_DB_USERNAME", mySqlContainer.getUsername());
+        System.setProperty("TEST_DB_PASSWORD", mySqlContainer.getPassword());
+    }
+
+    @Override
+    public void stop() {
```

Comment:
The stop() method is overridden but is empty. This could lead to resource leaks if the container is not properly stopped. You should call super.stop() inside the stop() method to ensure that the container is properly stopped.